src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/CreateCaseEventService.java

```diff
@@ -0,0 +1,233 @@
+    private final CaseDataService caseDataService;
+    private final CaseTypeService caseTypeService;
+    private final CaseSanitiser caseSanitiser;
+    private final CallbackInvoker callbackInvoker;
+    private final UIDService uidService;
+    private final SecurityClassificationService securityClassificationService;
+    private final ValidateCaseFieldsOperation validateCaseFieldsOperation;
+    private final UserAuthorisation userAuthorisation;
+
+    @Inject
+    public CreateCaseEventService(@Qualifier(CachedUserRepository.QUALIFIER) final UserRepository userRepository,
+                                  @Qualifier(CachedCaseDetailsRepository.QUALIFIER) final CaseDetailsRepository caseDetailsRepository,
+                                  @Qualifier(CachedCaseDefinitionRepository.QUALIFIER) final CaseDefinitionRepository caseDefinitionRepository,
+                                  final CaseAuditEventRepository caseAuditEventRepository,
+                                  final EventTriggerService eventTriggerService,
+                                  final EventTokenService eventTokenService,
+                                  final CaseService caseService,
+                                  final CaseDataService caseDataService,
+                                  final CaseTypeService caseTypeService,
+                                  final CaseSanitiser caseSanitiser,
+                                  final CallbackInvoker callbackInvoker,
+                                  final UIDService uidService,
+                                  final SecurityClassificationService securityClassificationService,
+                                  final ValidateCaseFieldsOperation validateCaseFieldsOperation,
+                                  final UserAuthorisation userAuthorisation) {
+        this.userRepository = userRepository;
+        this.caseDetailsRepository = caseDetailsRepository;
+        this.caseDefinitionRepository = caseDefinitionRepository;
+        this.caseAuditEventRepository = caseAuditEventRepository;
+        this.eventTriggerService = eventTriggerService;
+        this.caseService = caseService;
+        this.caseDataService = caseDataService;
+        this.caseTypeService = caseTypeService;
+        this.eventTokenService = eventTokenService;
+        this.caseSanitiser = caseSanitiser;
+        this.callbackInvoker = callbackInvoker;
+        this.uidService = uidService;
+        this.securityClassificationService = securityClassificationService;
+        this.validateCaseFieldsOperation = validateCaseFieldsOperation;
+        this.userAuthorisation = userAuthorisation;
+        mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+    }
+
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
+    public CreateCaseEventResult createCaseEvent(String caseReference, CaseDataContent content) {
+
+        final CaseDetails caseDetails = lockCaseDetails(caseReference);
```

Comment:
I've noticed this lock operation has been moved up a few lines. Any reason for that? Not sure but probably we could lock after retrieval of CaseType and CaseEvent? to keep the lock for less time

Reply:
it's in same position as before in DefaultCreateEventOperation. 
I guess, that's lock plus read (select for update operation). So, to retrieve CaseType and CaseEvent you need case-id.